test/test_loss.py

```diff
@@ -0,0 +1,82 @@
+"""Unit tests for the loss function."""
+from __future__ import print_function
+import json
+import numpy as np
+import tensorflow as tf
+from wavenet import make_loss, compute_receptive_field_size
+
+
+def make_logits():
+    # 1 batch item
+    # 5 time steps in duration
+    # 3 channels (quantization levels)
+    array = np.array([[[5.0,   -10.0, -10.0],
+                       [10.0,  -10.0, -10.0],
+                       [0.0,    10.0,   0.0],
+                       [-10.0,  10.0, -10.0],
+                       [-10.0,  10.0, -20.0]]])
+
+    return tf.convert_to_tensor(array, tf.float32)
+
+
+def make_labels():
+    # 1 batch item
+    # 5 time steps in duration
+    # 3 channels (quantization levels)
+    array = np.array([[[0.0, 1.0, 0.0],
+                       [0.0, 0.0, 1.0],
+                       [0.0, 0.0, 1.0],
+                       [0.0, 0.0, 1.0],
+                       [0.0, 0.0, 1.0]]])
+    return tf.convert_to_tensor(array, tf.float32)
+
+
+class TestLoss(tf.test.TestCase):
+    def setUp(self):
+        pass
+
+    def test_zero_receptive_field_size(self):
+        logits = make_logits()
+        labels = make_labels()
+
+        loss = make_loss(labels, logits, quantization_channels=3,
+                         receptive_field_size=0)
+        with self.test_session() as sess:
+            loss_array = sess.run([loss])
+        expected = np.array([[15.0, 20.0, 10.0, 20.0, 30.0]])
+        EPSILON = 0.001
+        np.testing.assert_allclose(expected, loss_array, rtol=EPSILON)
+
+    def test_non_zero_receptive_field_size(self):
+        logits = make_logits()
+        labels = make_labels()
+
+        loss = make_loss(labels, logits, quantization_channels=3,
+                         receptive_field_size=2)
+        with self.test_session() as sess:
+            loss_array = sess.run([loss])
+
+        # Since we truncate the first receptive_field time elements from the
+        # the tensors used to compute the loss, the losses should exclude the
+        # first two (since receptive_field_size is set to 2) that we would
+        # otherwise have.
+        expected = np.array([[10.0, 20.0, 30.0]])
+
+        EPSILON = 0.001
+        np.testing.assert_allclose(expected, loss_array, rtol=EPSILON)
+
+    def test_compute_receptive_field_size_4(self):
```

Comment:
Use a name that reflects the intention, not the implementation. `compute_receptive_field_size` should be give by the class name. Instead of `4`, use `_single_stack`